### PR TITLE
Re-vendor dependencies with GVT

### PR DESCRIPTION
**Pull request by jacobhaven**
Updating to use [GVT](https://github.com/FiloSottile/gvt) to remove some Godeps complexity

### CODE REVIEW

**Comment by jacobhaven** [comment]
Travis build errors are do to lack of c++11 support. https://github.com/travis-ci/travis-ci/issues/1379

File: .travis.yml [hunk 1]
```diff
@@ -44,6 +40,8 @@ before_script:
       psql -c 'create database certdb_development;' -U postgres;
       goose -path $GOPATH/src/github.com/cloudflare/cfssl/certdb/pg up;
     fi
+  # Install g++-4.8 to support c++11 for github.com/google/certificate-transparency/go/merkletree
+  - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test && sudo apt-get -qq update && sudo apt-get -qq install g++-4.8
```
Comment (jacobhaven) [inline .travis.yml]: @kisom Attempting to work-around be manually installing `g++-4.8`

**Comment by jacobhaven** [comment]
Testing against golang `tip` revealed a typo in `csr_test.go` and a bug in Bundler when bundling a chain with an expired intermediate (Go 1.6 throws an error Go 1.5 didn't).

**Comment by lziest** [comment]
TestRebundleFromPEM is also troublemaker :(

File: bundler/bundle_from_pem_test.go [hunk 2]
```diff
@@ -230,55 +230,54 @@ H5kVbpLP+eSYMqF110qqjAo4tkgBquF6IppA+HQ66DN64+TeiXb3f2Y=
...
+dqdkoIXvkmlScnu+6jBOaYeVvwogxr2Y69y4Zfg/qbPyBOLZquX9ovbuSP1DQmC/
+/LV5t7YHHY/1MXr5U0MMvcn+9JWUV6ou3at4AgEqfK0vAgMBAAGjZjBkMA4GA1Ud
+DwEB/wQEAwICBDASBgNVHRMBAf8ECDAGAQH/AgEBMB0GA1UdDgQWBBSIYLoYpHe4
+QQQb1e93UcJbFLogPzAfBgNVHSMEGDAWgBS4Xu+uZ1C31vMH5Wq+VbNnOg2SPjAN
+BgkqhkiG9w0BAQUFAAOBgQAnGnLG3r4g+0bLkeeh1Y71pL0Ui1LnvCA1v+yVDCd0
+G9pj7cnHHXjnp4Pic6pP9uwxBAiUC6rzjpKrm1ULYMoQPLAYmwJiz+8yiE5vpMCA
+Ov3LFPDNAbGF2wavwpCVolnVgHzPSFTEXN53DdXdVhcQ207P+zWNCNDF4Q33WSfm
+Dw==
 -----END CERTIFICATE-----
 -----BEGIN CERTIFICATE-----
```
Comment (jacobhaven) [inline bundler/bundle_from_pem_test.go]: @lziest Ugh, I generated this intermediate from `testdata/inter-L1.csr`, but the keys don't match!

**Comment by lziest** [comment]
LGTM

**Pull request merged**